spec/requests/topics_controller_spec.rb

```diff
@@ -5747,4 +5747,47 @@ def invite_group(topic, expected_status)
       end
     end
   end
+
+  describe ".defer_topic_view" do
+    fab!(:topic)
+    fab!(:user)
+
+    before do
+      Jobs.run_immediately!
+      Scheduler::Defer.async = true
+      Scheduler::Defer.timeout = 0.1
```

Comment:
Sorry this is not clear to be immediately but why do we need to set this to async?

Reply:
If we don't then the defer does not behave in the same way as production, and so the `return` issue doesn't manifest:

https://github.com/discourse/discourse/blob/289bbbde5a94ab067f0adfae81e7889b227cc6c8/lib/scheduler/defer.rb#L4-L10

I realised though I should set this back to async false after the spec, will do that